cdap-storage-ext-spanner/src/main/java/io/cdap/cdap/storage/spanner/SpannerStructuredTableAdmin.java

```diff
@@ -131,6 +110,45 @@ public StructuredTableSchema getSchema(StructuredTableId tableId) throws TableNo
     }
   }
 
+  private void updateTable(StructuredTableSpecification spec)
+    throws IOException, TableNotFoundException, TableSchemaIncompatibleException {
+    StructuredTableId tableId = spec.getTableId();
+    StructuredTableSchema cachedTableSchema = getSchema(tableId);
+    StructuredTableSchema newTableSchema = new StructuredTableSchema(spec);
+
+    if (newTableSchema.equals(cachedTableSchema)) {
+      LOG.trace("The table schema is already up to date: {}", tableId);
+      return;
+    }
+    if (!cachedTableSchema.isCompatible(newTableSchema)) {
+      throw new TableSchemaIncompatibleException(tableId);
+    }
+
+    List<String> statements = new ArrayList<>();
+    statements.addAll(getAddColumnsStatement(newTableSchema, cachedTableSchema));
+    statements.addAll(getAddIndicesStatement(newTableSchema, cachedTableSchema));
+    try {
+      Uninterruptibles.getUninterruptibly(adminClient.updateDatabaseDdl(databaseId.getInstanceId().getInstance(),
+                                                                        databaseId.getDatabase(), statements, null));
+      schemaCache.put(tableId, newTableSchema);
```

Comment:
I think it's better to invalidate the entry and let the next getSchema call to get the new table schema so that it is consistent with the rest of the code.

Reply:
good point!